src/engine/sys/sys_events.h

```diff
@@ -0,0 +1,140 @@
+/*
+===========================================================================
+Daemon BSD Source Code
+Copyright (c) 2017, Daemon Developers
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the Daemon developers nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL DAEMON DEVELOPERS BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+===========================================================================
+*/
+
+#ifndef ENGINE_SYS_SYS_EVENTS_H_
+#define ENGINE_SYS_SYS_EVENTS_H_
+
+#include <string>
+#include <vector>
+
+#include "common/Assert.h"
+#include "engine/qcommon/qcommon.h"
+
+namespace Sys {
+
+class EventBase {
+public:
+    const sysEventType_t type;
+    EventBase(sysEventType_t type) : type(type) { }
+
+    virtual ~EventBase() = 0;
```

Comment:
Choose if it is = 0 or defined below :P

Reply:
There's some weird rule with virtual destructors where you have to have both.

Reply:
I didn't know about this, do you have a reference?

Reply:
I should say, *pure* virtual destructors, to be precise. https://stackoverflow.com/a/21110040